app/middleware.go

```diff
@@ -17,8 +26,16 @@ func (app *App) handle(handler func(*framework.Response, *framework.Request)) ht
 }
 
 //RenderView renders a view
-func (app *App) renderView(viewHandler func(w http.ResponseWriter, req *http.Request)) http.HandlerFunc {
+func (app *App) renderView(viewHandler func(w http.ResponseWriter, req *http.Request), isSecurePage bool) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if isSecurePage {
```

Comment:
do you see this code being repeated all the time?Why not write a middleware for it?

Reply:
i tested it and it not repeated, well as long it's below line 30. When I ran it above the return (above line 30), I was repecting. I also run it in both positive and negative scenario and saw the code was running on page load once